### PR TITLE
Resolves gh-29: Adds a modulation matrix for editing different parameters of each layer.

### PR DESCRIPTION
**Pull request by colinbdclark**
This PR introduces a new, temporary user interface that allows artists to change different parameters of a video layer, including keyer and colour settings, playback speed, and opacity.

### CODE REVIEW

File: src/renderer-process/js/modulation-view.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        },
+
+        "afterRendered.applyBinding": {
+            funcName: "gpii.binder.applyBinding",
+            args: "{that}"
+        }
+    },
+
+    strings: {
+        id: "@expand:fluid.allocateGuid()",
```
Comment (amb26) [inline src/renderer-process/js/modulation-view.js]: You could just use your own id?

Comment (colinbdclark) [reply src/renderer-process/js/modulation-view.js]: As in, the component's ID supplied by Infusion? Or something else?

File: src/renderer-process/js/modulation-matrix-view.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+bubbles.modulationMatrixView.renderModViewContainers = function (that) {
+    fluid.each(that.options.strings, function (label, modulationName) {
+        bubbles.modulationMatrixView.renderModViewContainer(that,
+            modulationName, label);
+    });
+};
+
+bubbles.modulationMatrixView.renderModViewContainer = function (that, modulationName, label) {
+    var modViewContainer = $(that.options.markup.modulationViewContainer);
```
Comment (amb26) [inline src/renderer-process/js/modulation-matrix-view.js]: You may be interested to shift responsibility for markup and rendering into the modulationView component itself by means of the (somewhat undocumented) support for "fluid.containerRenderingView" in the new "protoframework" - https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/NewViewSupport.js#L57 - you can see a usage at https://github.com/fluid-project/infusion/blob/master/src/components/orator/js/Orator.js#L127

Comment (colinbdclark) [reply src/renderer-process/js/modulation-matrix-view.js]: Oooh, this is great—all of this "child container rendering" logic scattered around in my parent components has been really bothering me. Can this grade be used with mainline Infusion dev releases, or will I need to move to the potentia branch in order to use it?

File: src/renderer-process/js/video-layer-view.js [hunk 3]
```diff
@@ -71,9 +79,24 @@ fluid.defaults("bubbles.videoLayerView", {
...
         }
+    },
+
+    markup: {
+        modulationMatrixContainer: "<div class='bubbles-modulation-matrix'></div>"
     }
 });
+
+bubbles.videoLayerView.renderModulationMatrixContainer = function (that) {
+    var modulationMatrixContainer = $(that.options.markup.modulationMatrixContainer);
```
Comment (amb26) [inline src/renderer-process/js/video-layer-view.js]: See above for suggestions on "containerRenderingView" which will eliminate some complexity and an extra event

File: src/renderer-process/js/modulation-matrix-view.js (hunk 2 above)
Comment (amb26) [reply src/renderer-process/js/modulation-matrix-view.js]: The stuff linked is already in trunk, it's rickety (e.g. doesn't seem to have support for taking the markup back out again on "destroy" but then neither did you : P) but basically usable today

File: src/renderer-process/js/modulation-view.js (hunk 1 above)
Comment (amb26) [reply src/renderer-process/js/modulation-view.js]: Yes, simply {that}.id

File: src/renderer-process/js/modulation-matrix-view.js (hunk 2 above)
Comment (colinbdclark) [reply src/renderer-process/js/modulation-matrix-view.js]: That's so great, thanks for the heads up. It's nice to have a few supports for us self-rendering dinosaurs. :)

**Pull request merged**